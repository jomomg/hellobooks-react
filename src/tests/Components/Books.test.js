import React from "react";
import BookPage from "../../components/admin";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("<BookPage/>", ()=>{
    test("it renders successfully", ()=>{
        const books = shallow(<BookPage/>);
        expect(shallowToJson(books)).toMatchSnapshot();
    });
});