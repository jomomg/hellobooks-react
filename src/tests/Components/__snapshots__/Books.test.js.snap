// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookPage/> it renders successfully 1`] = `
<div>
  <TopNav
    title="Admin Dashboard"
  />
  <div
    style={
      Object {
        "marginTop": "3%",
      }
    }
  >
    <WithStyles(Paper)
      style={
        Object {
          "margin": "auto",
          "overflowX": "auto",
          "width": "90%",
        }
      }
    >
      <WithStyles(Table)>
        <WithStyles(TableHead)>
          <WithStyles(TableRow)>
            <WithStyles(TableCell)
              padding="dense"
            >
              Book ID
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Title
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Author
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Year
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Publisher
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Genre
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Actions
            </WithStyles(TableCell)>
          </WithStyles(TableRow)>
        </WithStyles(TableHead)>
        <WithStyles(TableBody) />
      </WithStyles(Table)>
    </WithStyles(Paper)>
    <DeleteDialog
      handleClose={[Function]}
      handleNo={[Function]}
      handleYes={[Function]}
      open={false}
    />
    <CreateBook
      author=""
      category=""
      description=""
      edition=""
      handleAddClick={[Function]}
      handleChange={[Function]}
      handleClose={[Function]}
      isbn=""
      open={false}
      publication_year=""
      publisher=""
      subcategory=""
      title=""
    />
  </div>
  <Notifier />
  <FloatingAddButton
    handleClick={[Function]}
  />
</div>
`;
